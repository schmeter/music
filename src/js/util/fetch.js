export const fetchJSON = (url, ignoreError = false) => new Promise((resolve, reject) => {
    fetch(url, {
        headers: {
            'Content-Type': 'application/json; charset=UTF-8'
        }
    })
        .then(res => {
            if (res.ok) {
                return res.json();
            }
            throw new Error(`${res.status} ${res.statusText}`);
        })
        .then(resolve)
        .catch(ignoreError ? console.error : reject); // eslint-disable-line no-console
});
