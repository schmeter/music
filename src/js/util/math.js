export const getRandom = (minValue, maxValue) => {
    let min = Math.min(minValue, maxValue);

    let max = Math.max(minValue, maxValue);

    if (min === max) {
        return (min);
    }
    return (min + parseInt(Math.random() * (max - min + 1), 10));
};
