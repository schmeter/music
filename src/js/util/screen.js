export const scrollTop = (querySelector) => {
    const container = document.querySelectorAll(querySelector)[0];
    if (container) {
        container.scrollTop = 0;
    }
};

export const isTouch = () => !!('ontouchstart' in window);
