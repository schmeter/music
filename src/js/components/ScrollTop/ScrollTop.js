import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import { scrollTop } from '../../util/screen';

class ScrollTop extends React.Component {
    componentDidUpdate(lastProps) {
        const { location, target, onReRender } = this.props;
        const { location: lastLocation } = lastProps;
        const shouldScrollTop = (typeof onReRender === 'boolean' && onReRender)
            || lastLocation.pathname !== location.pathname;

        if (shouldScrollTop) {
            scrollTop(target);
        }
    }

    render() {
        const { children } = this.props;

        return React.Children.only(children);
    }
}

ScrollTop.propTypes = {
    location: PropTypes.shape({
        pathname: PropTypes.string.isRequired
    }).isRequired,
    children: PropTypes.node.isRequired,
    target: PropTypes.string.isRequired,
    onReRender: PropTypes.bool
};

export default withRouter(ScrollTop);
