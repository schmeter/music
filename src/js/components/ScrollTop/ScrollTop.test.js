import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ScrollTop from './ScrollTop';

import { scrollTop } from '../../util/screen';

jest.mock('../../util/screen.js', () => ({ scrollTop: jest.fn() }));

describe('ScrollTop', () => {
    afterEach(() => {
        scrollTop.mockReset();
    });

    it('renders the children', () => {
        const target = '.foo';
        const component = shallow(
            <ScrollTop target={target}>
                test
            </ScrollTop>
        );
        expect(toJson(component)).toMatchSnapshot();
    });

    it('scrolls to top on re-render', () => {
        const target = '.ipsum';
        const component = shallow(
            <ScrollTop target={target}>
                test
            </ScrollTop>
        );

        component.setProps({
            children: 'foo'
        });

        expect(scrollTop).toHaveBeenCalledWith(target);
    });
});
