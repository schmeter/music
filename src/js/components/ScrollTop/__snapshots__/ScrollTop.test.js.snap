// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollTop renders the children 1`] = `
<Route>
  <Component />
</Route>
`;
